src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/CvlRecordService.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.dates.ReleaseDateService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+
+@Service
+class CvlRecordService(
```

Comment:
This class is the crux of the changes. In almost all cases, the eligibility service is no longer called directly, nor is `getLicenceStartDate(s)`. The only places that call the eligibility service directly are those where licences already exist, meaning LicenceStartDate does not need to be recalculated.
By utilising this with the recently eligibility service refactor, we no longer need to have `determineReleaseDateKind`, as it is replaced by `eligibleKind`. This in turn means we no longer have two places where eligibility is calculated (for `NOT_STARTED` cases via the eligibility service, and on-creation via the licence creation service)